OrbitLinuxTracing/GpuTracepointEventProcessor.h

```diff
@@ -0,0 +1,87 @@
+
+#ifndef ORBIT_LINUX_TRACING_GPU_TRACEPOINT_EVENT_PROCESSOR
+#define ORBIT_LINUX_TRACING_GPU_TRACEPOINT_EVENT_PROCESSOR
+
+#include <tuple>
+#include "OrbitLinuxTracing/TracerListener.h"
+#include "PerfEvent.h"
+#include "absl/container/flat_hash_map.h"
+
+namespace LinuxTracing {
+
+class GpuTracepointEventProcessor {
+ public:
+  GpuTracepointEventProcessor(int amdgpu_cs_ioctl_id,
+                              int amdgpu_sched_run_job_id,
+                              int dma_fence_signaled_id)
+      : amdgpu_cs_ioctl_id_(amdgpu_cs_ioctl_id),
+      amdgpu_sched_run_job_id_(amdgpu_sched_run_job_id),
+      dma_fence_signaled_id_(dma_fence_signaled_id) {}
+
+  void PushEvent(const std::unique_ptr<PerfEventSampleRaw>& sample);
+  void SetListener(TracerListener* listener);
+
+ private:
+  // Keys are context, seqno, and timeline
+  typedef std::tuple<uint32_t, uint32_t, std::string> Key;
+
+  template<typename T> std::string ExtractTimelineString(
+      const T* tracepoint_data) {
+    int32_t data_loc = tracepoint_data->timeline;
+    int16_t data_loc_size = static_cast<int16_t>(data_loc >> 16);
+    int16_t data_loc_offset = static_cast<int16_t>(data_loc & 0x00ff);
+
+    std::vector<char> data_loc_data(data_loc_size);
+    std::memcpy(&data_loc_data[0],
```

Comment:
Is data_loc_data guaranteed to be null terminated?  Should we punch in the null character to be sure?

Reply:
It is null terminated, but I'm not sure this is guaranteed. Added a comment and code to punch in the zero (which is a no-op in case we get what we expect).

Reply:
Cool!